consensus/istanbul/backend/api.go

```diff
@@ -474,6 +477,100 @@ func (api *APIExtension) GetBlockWithConsensusInfoByHash(blockHash common.Hash)
 	return api.makeRPCBlockOutput(block, cInfo, block.Transactions(), receipts), nil
 }
 
+func (api *APIExtension) GetBlsInfos(number rpc.BlockNumber) (map[string]interface{}, error) {
+	kip113Addr, err := api.GetActiveAddressFromRegistry(system.Kip113Name, number)
+	if err != nil {
+		return nil, err
+	}
+
+	var bn *big.Int
+	if number == rpc.LatestBlockNumber || number == rpc.PendingBlockNumber {
+		bn = big.NewInt(api.chain.CurrentBlock().Number().Int64())
+	} else {
+		bn = big.NewInt(number.Int64())
+	}
+
+	backend := backends.NewBlockchainContractBackend(api.chain, nil, nil)
+	if infos, err := system.ReadKip113All(backend, kip113Addr, bn); err != nil {
+		return nil, err
+	} else {
+		blsInfos := make(map[string]interface{})
+		// make publicKey, Pop to hexadecimal string
+		for addr, info := range infos {
+			blsInfos[addr.Hex()] = map[string]interface{}{
+				"publicKey": hex.EncodeToString(info.PublicKey),
+				"pop":       hex.EncodeToString(info.Pop),
+				"verifyErr": info.VerifyErr,
+			}
+		}
+		return blsInfos, nil
+	}
+}
+
+func (api *APIExtension) GetAllRecordsFromRegistry(name string, number rpc.BlockNumber) ([]interface{}, error) {
+	var bn *big.Int
+	if number == rpc.LatestBlockNumber || number == rpc.PendingBlockNumber {
+		bn = big.NewInt(api.chain.CurrentBlock().Number().Int64())
+	} else {
+		bn = big.NewInt(number.Int64())
+	}
+
+	if api.chain.Config().IsRandaoForkBlock(bn) && name == system.Kip113Name {
```

Comment:
At Randao fork block, KIP113 must have been registered. So I think we're able to return the correct value without this branch, given a hardfork block + kip113 request.

Reply:
We need this branch since `Registry` will be deployed at `RandaoHFBlock`, not `RandaoHFBlock - 1`. We can read `Registry` from `RandaoHFBlock + 1`.

Reply:
We need to fix wrong comments here: https://github.com/klaytn/klaytn/blob/dev/consensus/istanbul/backend/engine.go#L548. Do you think it's appropriate to fix this in this PR?

Reply:
It's installed at randao fork, so it's viewable:

```
$ cast call 0x0000000000000000000000000000000000000401 "getAllRecords(string)((address,uint256)[])" KIP113 -r baobab -b 141366999
Error:
Contract 0x0000000000000000000000000000000000000401 does not exist

$ cast call 0x0000000000000000000000000000000000000401 "getAllRecords(string)((address,uint256)[])" KIP113 -r baobab -b 141367000
[(0x4BEed0651C46aE5a7CB3b7737345d2ee733789e6,0)]
```

Reply:
yep, please fix the comment in this PR as well

Reply:
Fix wrong comments applied at https://github.com/klaytn/klaytn/pull/2111/commits/88f01d7bdc8dbf2f977b2b9718e96034ed8aa00e

Fix useless branch at https://github.com/klaytn/klaytn/pull/2111/commits/faa74df00007e1bae023a3b87b8581171db91d24